bench/run_bench.py

```
@@ -0,0 +1,445 @@
+    _PERF = shutil.which("perf")
+    COUNTER_BACKEND = "perf" if _PERF else "none"
+
+    def perf_window(pid, cmd):
+        """instructions, cycles for `pid` while `cmd` runs, plus elapsed seconds.
+
+        `perf stat -p PID -- CMD` attaches to PID, runs CMD, and stops counting
+        when CMD exits, so the counters cover exactly the benchmark window.
+        `-x,` gives machine-readable `value,unit,event,...` lines on stderr.
+        """
+        t0 = time.time()
+        out = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)